cpp/src/parquet/util/crypto.h

```diff
@@ -18,29 +18,126 @@
 #ifndef PARQUET_UTIL_CRYPTO_H
```

Comment:
Is PARQUET_EXPORT needed on any functions in here? If might be clearer to move this file to the top level of the Parquet codebase and call it `encryption-internal.h` if it is internal-only